types/node/v8/node-tests.ts

```diff
@@ -819,6 +819,7 @@ namespace util_tests {
         var arg0NoResult: () => Promise<any> = util.promisify((cb: (err: Error) => void): void => { });
         var arg1: (arg: string) => Promise<number> = util.promisify((arg: string, cb: (err: Error, result: number) => void): void => { });
         var arg1NoResult: (arg: string) => Promise<any> = util.promisify((arg: string, cb: (err: Error) => void): void => { });
+        var cbOptionalError: () => Promise<void> = util.promisify((cb: (err?: Error | null) => void): void => { cb(); });
```

Comment:
I don't know why but you can avoid the typescript 2.3 error by using `util.promisify<void>`. Seems typescript 2.3 selects the wrong overload.

Reply:
Hmm... I tried your suggestion, but it still chose the wrong overload:

```
node/v8 failing:
Error: /Users/jstewmon/dev/forks/DefinitelyTyped/types/node/v8/node-tests.ts:822:52
ERROR: 822:52  no-void-expression  Expression has type `void`. Put it on its own line as a statement.
ERROR: 822:73  expect              TypeScript@next compile error:
Argument of type '(cb: (err?: Error) => void) => void' is not assignable to parameter of type '(arg1: void, callback: (err?: Error) => void) => void'.
  Types of parameters 'cb' and 'arg1' are incompatible.
    Type 'void' is not assignable to type '(err?: Error) => void'.
```

I think I see why that didn't work - giving a type parameter is going to limit the choices to `util.promisify<TResult>` or `util.promisify<T1>`, but the fn argument satisfies neither of those (neither takes an argument nor accepts callback with two arguments).

Is there way to hint to the compiler that it should choose the non-generic overload?

Reply:
For efficiency we normally only test against the oldest and newest versions unless there's an error.
It looks like you can fix this in ts2.3 by adding a second parameter `, result?: void`.

Reply:
@andy-ms adding a second parameter would invalidated the test b/c it will cause `util.promisify<TResult>` to be the selected overload, and that particular overload is not affected by this change.

Reply:
I looked at this for a while and couldn't find any change that wouldn't make things worse in `typescript@next`, which I think is more important. Probably better to comment out this test for now and note that it works in `typescript@2.4` and up.

Reply:
Ok, I'll make that change. Thanks for investigating.

Is there a process for selecting the minimum version to test in the header? e.g. When global usage of version x falls below y.

I have no idea how often the version compatibility issue comes up, but it would be great if we could have min-version-bound tests for each package to selectively run tests for features that are supported from version <x>+.

Reply:
Er... as I wrote that I realized the definitions are where the version constraint is currently indicated, not in the tests...